layouts/shortcodes/flexcolumns.html

```diff
@@ -0,0 +1,22 @@
+{{ $mobile := "" }}
```

Comment:
Something I find it useful to do for shortcodes is get/declare the variables at the top of the file to help clarify what the shortcode can expect to receive from a user / allow us to just reference the variables further down (unless they have to be pulled from the current scope, like `$ratioSplit` does). That also lets us handle when the shortcodes are called without named parameters.

You can such an example here:

https://github.com/platenio/hugo-platen/blob/64d2e468e33b53a3c14130bfe3561ba92bea5f36/layouts/shortcodes/itchio.html#L1-L11

We can _also_ use the [`default` function](https://gohugo.io/functions/default/) to set any defaults we need during parameter declaration. We could possibly use this mechanism to define a default ratio to keep from forking the code later (or not, that may not make sense here).

Reply:
I had no idea about .IsNamedParams or scratch! I've refactored to do this as well as use both of these, those are really good options.